### PR TITLE
Bump golangci-lint for parity with CI and increase static analysis timeout

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Bump golangci-lint to v1.57.2 for parity with CI and increase the static analysis timeout to 10 min

```
[15:32:18] ~/lab/gloo ➦ bump-linter $ make print-LINTER_VERSION
v1.57.2
[15:32:25] ~/lab/gloo ➦ bump-linter $ make install-go-tools
...
go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.57.2
...
```

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sam-heilbron — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -148,7 +148,7 @@ install-go-tools: mod-download ## Download and install Go dependencies
 	go install github.com/golang/mock/mockgen
 	go install github.com/saiskee/gettercheck
 	# This version must stay in sync with the version used in CI: .github/workflows/static-analysis.yaml
-	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.54.2
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.57.2
```
Comment (nfuden) [inline Makefile:151]: nit: any way to keep this in lock step rther than having to manually bump given we have this claimed one to one dep?

**Review by nfuden — APPROVED** [review]
Left a nit. That nit is now in the authors hands

File: Makefile (hunk 1 above)
Comment (davidjumani) [reply Makefile]: Updated to read the version from the static analysis GHA

**Review by sam-heilbron — APPROVED** [review]

**Review by jenshu — APPROVED** [review]

**Pull request merged**